front/src/views/sign-in/sign-in.view.js

```diff
@@ -29,6 +30,12 @@ export default {
     },
```

Comment:
```
.then(() => {
          this.$router.push('/');
        }).catch(() => {
          this.serverError = 'Invalid credentials';
        });
```

so, it means that your logical handling of authorization depends on requirements of controller. because your controller is sure that it was invalid credentionals

it is more pure to return from action some flags based on code of response. also you could compare status code

it is a bit confuced for me to see serverError which is setted by const string in all rejected states